modules/olm-overriding-operator-pod-affinity.adoc

```diff
@@ -0,0 +1,202 @@
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  channel: stable
+  config:
+    affinity:
+      nodeAffinity: <1>
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+                - values:
+                    - ip-10-0-163-94.us-west-2.compute.internal
+                  key: kubernetes.io/hostname
+                  operator: In
+ ...
+----
+<1> A node affinity that requires the Operator's pod to be scheduled on a node named `ip-10-0-163-94.us-west-2.compute.internal`.
+
+.Node affinity example that places the Operator pod on a node with a specific architecture
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  channel: stable
+  config:
+    affinity:
+      nodeAffinity: <1>
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: kubernetes.io/arch
+              operator: In
+              values:
+              - arm64
+            - key: kubernetes.io/os
+              operator: In
+              values:
+                - linux
+----
+<1> A node affinity that requires the Operator's pod to be scheduled on a node with the `kubernetes.io/arch=arm64` and `kubernetes.io/os=linux` labels.
+
+.Pod anti-affinity example that prevents the Operator pod from one or more specific nodes
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  creationTimestamp: '2022-12-21T14:54:47Z'
+  generation: 6
+  labels:
+    operators.coreos.com/openshift-custom-metrics-autoscaler-operator.openshift-keda: ''
+  name: openshift-custom-metrics-autoscaler-operator
+  namespace: openshift-keda
+  resourceVersion: '77887'
+  uid: 9b38d0eb-9c57-4cb6-9905-1c65c21f08ef
+spec:
+  channel: stable
+  config:
+    affinity:
+      podAntiAffinity: <1>
+        requiredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+          matchExpressions:
+            - key: kubernetes.io/arch
+              operator: In
+              values:
+              - amd64
+          topologyKey: kubernetes.io/hostname
+ ...
+----
+<1> A pod anti-affinity that prevents the Operator's pod from being scheduled on a node with the `kubernetes.io/arch=amd4` label.
```

Comment:
This is not correct.

Pod anti-affinity doesn't check node labels, it checks pod labels for pods running on that node.
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity

If you want to avoid a node based on architecture node labels, you'd use node anti affinity.

Reply:
Technically, there is no separate node anti affinity. You'd just use node affinity with negative selection rules. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity for a full explanation.

Reply:
@jaypoulz What is a good use case for using pod antiaffinity when installing an Operator?

Reply:
Say you had two workloads that are intensive on a particular resource. Let's just say CPU, as an example. One thing you might way to do is to ensure they aren't scheduled on the same node. Pod anti affinity allows you to say - check the scheduled pods on this node, and make sure there are none that match this label (or set of labels)

Essentially, you only care that something is not scheduled in the same place of something that could conflict with it.